devicetypes/smartthings/zwave-siren-tamper.src/zwave-siren-tamper.groovy

```diff
@@ -0,0 +1,283 @@
+ *  for the specific language governing permissions and limitations under the License.
+ */
+metadata {
+	definition(name: "Z-Wave Siren Tamper", namespace: "smartthings", mnmn: "SmartThings", vid: "generic-siren-tamper", author: "SmartThings", ocfDeviceType: "x.com.st.d.siren", runLocally: true, minHubCoreVersion: '000.017.0012', executeCommandsLocally: false) {
+		capability "Actuator"
+		capability "Alarm"
+		capability "Battery"
+		capability "Refresh"
+		capability "Switch"
+		capability "Tamper Alert"
+
+		command "test"
+		//zw:Fs type:1005 mfr:0129 prod:6F01 model:0001 ver:1.04 zwv:4.33 lib:03 cc:5E,80,5A,72,73,86,70,98 sec:59,2B,71,85,25,7A role:07 ff:8F00 ui:8F00
+		fingerprint mfr: "0129", prod: "6F01", model: "0001", deviceJoinName: "Yale External Siren"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name: "alarm", type: "generic", width: 6, height: 4) {
+			tileAttribute("device.alarm", key: "PRIMARY_CONTROL") {
+				attributeState "off", label: 'off', action: 'alarm.siren', icon: "st.alarm.alarm.alarm", backgroundColor: "#ffffff"
+				attributeState "both", label: 'alarm!', action: 'alarm.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#e86d13"
+			}
+		}
+		standardTile("test", "device.alarm", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: '', action: "test", icon: "st.secondary.test"
+		}
+		standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: '', action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+		valueTile("tamper", "device.tamper", decoration: "flat", width: 2, height: 2) {
+			state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+			state "detected", label: 'tampered', backgroundColor: "#ff0000"
+		}
+
+		preferences {
+			input name: "alarmLength", type: "number", title: "Alarm length (1-10 min)", range: "1..10"
+			// defaultValue: 10
+			input name: "alarmLEDflash", type: "bool", title: "Alarm LED flash"
+			// defaultValue: false
+			input name: "comfortLED", type: "number", title: "Comfort LED (0-25 x 10 sec.)", range: "0..25"
+			// defaultValue: 0
+			input name: "tamper", type: "bool", title: "Tamper alert"
+			// defaultValue: false
+		}
+
+		main "alarm"
+		details(["alarm", "test", "off", "refresh", "battery", "tamper"])
+	}
+}
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
```

Comment:
For a battery-powered device, this should be longer. `zwave-siren` uses 122 minutes